terraform/cluster.tf

```diff
@@ -25,6 +25,37 @@ resource "google_container_cluster" "primary" {
       ip_allocation_policy
     ]
   }
+
+  /**
```

Comment:
To keep the code clean, this should be documented in the README.

Reply:
Comments in code tend to go stale or worse not reflect the actual code. Where the code isn't self descriptive, we should fall back to documenting it in the relevant readme section.

Reply:
Trace, really appreciate your review here! 🙏 

Just so I understand, are you really advocating that ALL inline code comments are a practice to avoid? You yourself have left inline comments in this [same file](https://github.com/pangeo-forge/pangeo-forge-gcs-bakery/blame/main/terraform/cluster.tf#L51). I don't see how these are qualitatively different.

Whether code is self-descriptive depends a lot on the audience. Charles and I are completely new to terraform, so some inline comments can be really helpful in understanding what a particular code block is for. I agree there is a maintenance cost to adding comments.

Maybe rather than saying,  "let's avoid all inline comments," we could ask, "are these specific notes better placed inline vs. in the README?"

Reply:
This is a fair counter-point. I need to look into more around best practises when it comes to Terraform and code comments, but I have always been told "inline comments bad" across the languages I have written in several companies.

Unfortunately, the comment which was in that file was as a result of a shameless copy-pasta on my part, rather than an intentional "This comment needs to be here to inform people". And this particular one should also be migrated to the README, which I will take an action point to do.

I'd prefer to retain the reduction of inline comments where possible, and perhaps flesh out the "per file" sections of the readme with references to the codeblocks where these are needed to help newer contributors. 
This has a few benefits:
- Keeping the code uncluttered
- Allowing ease of searching (versus GitHubs code search which is buggy at best)
- Keeping all the information in one place - Making the README the Goto for "How does this thing work"

Reply:
WRT the above: it looks like I was thinking of another repo where each file and its purpose was in the README. I will take an action item to get this added to the README for this.

Reply:
Ok great, so it sounds like we have a path forward.
- Let's finish up this review focusing on the actual code and get it merged so Charles can unblock his next steps
- Next you can make a pass through and migrate inline comments into a Readme to suit your style preferences (which are certainly better-informed than ours)

Does that sound oK?

Reply:
That sounds spot on. 

The main reason for requesting the changes was to prompt the "Is this something we can do very quickly? If not, then when?" discussion. And I agree that "Soon, but not now" is the right answer. 

If you are happy with the approach RE: explaining each file and its purpose in the readme (plus a section for gotchas maybe?), then that sounds good to me.